internal/aws.go

```diff
@@ -96,10 +111,7 @@ func assumeRoleFromAWS(arn string, request *Request) (*sts.AssumeRoleOutput, err
 
 	request.setLabel("assume_role_from_aws_cache", "miss")
 	request.log.Infof("Requesting STS Assume Role info for %s from AWS", arn)
-	req := stsService.AssumeRoleRequest(&sts.AssumeRoleInput{
-		RoleArn:         aws.String(arn),
-		RoleSessionName: aws.String("go-metadataproxy"),
-	})
+	req := stsService.AssumeRoleRequest(constructAssumeRoleInput(arn, externalId, "go-metadataproxy"))
```

Comment:
Would move `go-metadataproxy` into being hardcoded in the `RoleSessionName` since its static and users can't change it anyway, fewer arguments to pass around=win

```suggestion
	req := stsService.AssumeRoleRequest(constructAssumeRoleInput(arn, externalId))
```